src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -239,7 +251,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       commitUnmount(node);
       // Visit children because they may contain more composite or host nodes.
       // Skip portals because commitUnmount() currently visits them recursively.
-      if (node.child !== null && node.tag !== HostPortal) {
+      if (
+        node.child !== null &&
+        // Drill down into portals only if we use mutation since that branch is recursive
+        (!mutation || node.tag !== HostPortal)
```

Comment:
Doesn't this comment lie about what it does?

Reply:
yea

Reply:
Either it will either have already drilled down in commitUnmount or it will let it drill further here.

Reply:
Fixed.